### PR TITLE
added a pre-order traversal of the scene graph.

### PR DESCRIPTION
**Pull request by shamanDevel**
Adds a new method called depthLastTraversal() to Spatial and subclasses as an extension to depthFirstTraversal() and breathFirstTraversal().

### CODE REVIEW

**Comment by empirephoenix** [comment]
So any answer here? This is kinda long stalled as well, but look sfine for me, merge?

**Comment by empirephoenix** [comment]
@shadowislord  @pspeed42 Any rejections? Else I will merge this next time I review the open pull requests.

**Comment by Shredder121** [comment]
Is it maybe a good idea to **add** a method rather than replacing it?
Backward compatible patches are more likely to be applied quickly.

File: jme3-core/src/main/java/com/jme3/scene/Spatial.java [hunk 1]
```diff
@@ -1492,10 +1492,23 @@ public Matrix4f getLocalToWorldMatrix(Matrix4f store) {
...
     /**
-     * Visit each scene graph element ordered by DFS
+     * Visit each scene graph element ordered by DFS, also called post order traversal.
+     * Delegates to {@link #depthFirstTraversal(com.jme3.scene.SceneGraphVisitor, boolean) }
+     * with {@code preOrder} set to {@code false}.
      * @param visitor
      */
-    public abstract void depthFirstTraversal(SceneGraphVisitor visitor);
+    public void depthFirstTraversal(SceneGraphVisitor visitor) {
+        depthFirstTraversal(visitor, false);
```
Comment (Shredder121) [inline jme3-core/src/main/java/com/jme3/scene/Spatial.java]: I didnt see this one, my comment can be ignored.

**Comment by empirephoenix** [comment]
Closing due to merge conflict, however I would like to see this again in a form that just adds methods, as that can be easily applies without braking anything. I personally find different traversal methods usefull.

**Comment by shamanDevel** [comment]
Oops, I totally forgot that this was still hanging around.
I'll start with a clean pull from the head of the jme3 branch again. Then there should be no merge conflicts and I'll include all the suggestions. You'll see a new pull request soon.

**Pull request closed without merging**